### ISSUE TITLE
Uninformative error message - Issue with Windows Build

### ISSUE THREAD

**Issue by adam2392** [pypa/cibuildwheel#1748]
### Description

Hi,

Thank you for this great package. I'm having a bit of trouble building wheels all of a sudden (https://github.com/neurodata/scikit-tree/pull/207), but it previously works: https://github.com/neurodata/scikit-tree/actions/runs/7721644419/job/21048502802.

I was wondering if anyone had any idea on how to debug this even, or if there was a recent bug introduced?

```
Run pipx run --python "C:\hostedtoolcache\windows\Python\3.12.1\x64\python.exe" --spec "D:\a\_actions\pypa\cibuildwheel\v2.16.4" cibuildwheel "." --output-dir '"wheelhouse"' --config-file '""' --only '""'
  pipx run --python "C:\hostedtoolcache\windows\Python\3.12.1\x64\python.exe" --spec "D:\a\_actions\pypa\cibuildwheel\v2.16.4" cibuildwheel "." --output-dir '"wheelhouse"' --config-file '""' --only '""'
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    IS_32_BIT: false
    pythonLocation: C:\hostedtoolcache\windows\Python\3.9.13\x64
    PKG_CONFIG_PATH: C:\hostedtoolcache\windows\Python\3.9.13\x64/lib/pkgconfig
    Python_ROOT_DIR: C:\hostedtoolcache\windows\Python\3.9.13\x64
    Python2_ROOT_DIR: C:\hostedtoolcache\windows\Python\3.9.13\x64
    Python3_ROOT_DIR: C:\hostedtoolcache\windows\Python\3.9.13\x64
    CIBW_BUILD: cp311-win*
    CIBW_ARCHS: AMD64
    CIBW_ENVIRONMENT_PASS_LINUX: RUNNER_OS
    CIBW_ENVIRONMENT_WINDOWS: PKG_CONFIG_PATH=c:/opt/64/lib/pkgconfig PIP_PRE=1 PIP_EXTRA_INDEX_URL=https://pypi.anaconda.org/scientific-python-nightly-wheels/simple PIP_NO_BUILD_ISOLATION=false
  
Invalid --only='""', must be a build selector with a known platform
```

### Build log

https://github.com/neurodata/scikit-tree/actions/runs/7730700062/job/21076674827?pr=207

### CI config

https://github.com/neurodata/scikit-tree/blob/main/.github/workflows/build_wheels.yml

**Comment by bhazelton** [comment]
I believe this was fixed in v2.16.5. I'm still having some new build issues in Windows on that version, but it's a different failure mode so may not affect you.

**Comment by henryiii** [comment]
Yes, that was due to a GitHub rollout of new builder images that contained Powershell 7.3+, which changed the way quoting works. It's way better, but the weird quoting required for 7.2 and earlier causes quotes to be injected to both `--only` as well as the paths. We've fixed it in 2.16.5.

**Comment by henryiii** [comment]
I've updated the title of #1740 and pinned it, maybe that will help.
